### PR TITLE
Make sure correlation dummy span use integers for trace id and span id

### PR DESCRIPTION
**Pull request by DylanLovesCoffee**
### What does this PR do?

Converts trace ID and span ID from the DD trace context headers into integers.

### Motivation

Fixes a TypeError raised by dd-trace-py: #134 when `DD_TRACE_ENABLED=false`.

With dd-trace-py v0.48.0, a [type check was implemented](https://github.com/DataDog/dd-trace-py/pull/2243) to make sure trace and span IDs are integers. Our trace ID and span IDs were not typed as integers and would raise an exception crashing the function.

### Testing Guidelines

Fixed unit test + built the layer locally to test.

### Types of Changes

- [x] Bug fix
- [ ] New feature
- [ ] Breaking change
- [ ] Misc (docs, refactoring, dependency upgrade, etc.)

### Check all that apply

- [x] This PR's description is comprehensive
- [ ] This PR contains breaking changes that are documented in the description
- [ ] This PR introduces new APIs or parameters that are documented and unlikely to change in the foreseeable future
- [ ] This PR impacts documentation, and it has been updated (or a ticket has been logged)
- [ ] This PR's changes are covered by the automated tests
- [ ] This PR collects user input/sensitive content into Datadog
- [ ] This PR passes the integration tests (ask a Datadog member to run the tests)

### CODE REVIEW

**Review by hghotra — APPROVED** [review]

File: tests/integration/snapshots/return_values/sync-metrics_python38_sqs.json [hunk 1]
```diff
@@ -1,3 +1,4 @@
+Serverless: Running "serverless" installed locally (in service node_modules)
```
Comment (tianchu) [inline tests/integration/snapshots/return_values/sync-metrics_python38_sqs.json:1]: What is this? Can we remove it? or mute it in serverless framework somehow?

File: tests/integration/snapshots/logs/sync-metrics_python36.log [hunk 2]
```diff
@@ -1,24 +1,25 @@
...
...9012:my-queue"}, "metrics": {"_sampling_priority_v1": 1}, "type": "serverless"}, {"trace_id": "XXXX", "parent_id": "XXXX", "span_id": "XXXX", "service": "aws.lambda", "resource": "requests.request", "name": "requests.request", "error": 0, "start": XXXX, "duration": XXXX, "meta": {"_dd.origin": "lambda", "http.method": "GET", "http.url": "https://httpstat.us/200/", "http.status_code": "200"}, "metrics": {"_dd.measured": 1}, "type": "http"}, {"trace_id": "XXXX", "parent_id": "XXXX", "span_id": "XXXX", "service": "aws.lambda", "resource": "requests.request", "name": "requests.request", "error": 0, "start": XXXX, "duration": XXXX, "meta": {"_dd.origin": "lambda", "http.method": "GET", "http.url": "https://httpstat.us/400/", "http.status_code": "400"}, "metrics": {"_dd.measured": 1}, "type": "http"}, {"trace_id": "XXXX", "parent_id": "XXXX", "span_id": "XXXX", "service": "aws.lambda", "resource": "requests.request", "name": "requests.request", "error": 0, "start": XXXX, "duration": XXXX, "meta": {"_dd.origin": "lambda", "http.method": "POST", "http.url": "https://api.datadoghq.com/api/v1/distribution_points", "http.status_code": "202"}, "metrics": {"_dd.measured": 1}, "type": "http"}]]}
```
Comment (tianchu) [inline tests/integration/snapshots/logs/sync-metrics_python36.log:23]: Not sure it's a big deal, but do you know why the `runtime-id` is gone from the snapshot?

File: tests/integration/snapshots/return_values/sync-metrics_python38_sqs.json (hunk 1 above)
Comment (DylanLovesCoffee) [reply tests/integration/snapshots/return_values/sync-metrics_python38_sqs.json:1]: Yeah this one is weird. Looking into it 🤔

File: tests/integration/snapshots/logs/sync-metrics_python36.log (hunk 2 above)
Comment (DylanLovesCoffee) [reply tests/integration/snapshots/logs/sync-metrics_python36.log:22]: I ran the integration tests for the previous release and I get a mismatch error where the `runtime-id` is missing in some places too. Looks to be an issue within dd-trace-py we might want to dig deeper into.

File: tests/integration/snapshots/return_values/sync-metrics_python38_sqs.json (hunk 1 above)
Comment (DylanLovesCoffee) [reply tests/integration/snapshots/return_values/sync-metrics_python38_sqs.json]: Had to downgrade my version of serverless to fix this. Seems like the latest versions of the serverless framework will print that log out without some sort of `--quiet` flag.

Comment (tianchu) [reply tests/integration/snapshots/return_values/sync-metrics_python38_sqs.json]: https://github.com/DataDog/datadog-lambda-python/blob/b0158b6f4315a197ed3044f60df35e786cc3bc1b/scripts/run_integration_tests.sh#L144 you can actually remove them this way, but it's ok we can deal with it next time.

**Pull request merged**